modules/jupyter-widget/src/widget.js

```diff
@@ -60,53 +49,52 @@ export class DeckGLModel extends DOMWidgetModel {
   }
 }
 
-const TICK_RATE_MILLISECONDS = 100;
-
 export class DeckGLView extends DOMWidgetView {
   render() {
-    this.modelId = this.model.model_id;
     super.render();
     this.model.on('change:json_input', this.value_changed, this);
     loadCss(MAPBOX_CSS_URL);
-    const [width, height] = [this.model.get('width'), this.model.get('height')];
 
-    createDeckScaffold(this.el, this.modelId, width, height);
+    this.el.id = this.model.model_id;
 
-    waitForElementToDisplay(
-      `#deck-map-wrapper-${this.modelId}`,
-      TICK_RATE_MILLISECONDS,
-      this.initJSElements.bind(this)
-    );
+    createWidgetDiv({
+      parentDiv: this.el,
+      idName: this.model.model_id,
+      done: this.initDeckElements.bind(this),
+      width: this.model.get('width'),
+      height: this.model.get('height')
+    });
   }
 
   _onViewStateChange({viewState}) {
     this.deck.setProps({viewState});
-    setMapProps(this.mapLayer, {viewState});
   }
 
-  initJSElements() {
+  initDeckElements() {
+    setDependencies(this._initDeck.bind(this));
+  }
+
+  _initDeck(deckgl, mapboxgl) {
     try {
       if (!this.deck) {
-        this.deck = new Deck({
-          canvas: `deck-map-container-${this.modelId}`,
+        this.deck = new deckgl.Deck({
```

Comment:
[DeckGL](https://deck.gl/#/documentation/deckgl-api-reference/scripting-interface/deckgl)

Reply:
Changed in latest commit